### ISSUE TITLE
ChatCompletionMessageParam - there is an TS error when trying to use content as an object

### ISSUE THREAD

**Issue by arekrzebit** [openai/openai-node#867]
### Confirm this is a Node library issue and not an underlying OpenAI API issue

- [X] This is an issue with the Node library

### Describe the bug

When we use OpenAI.chat.completions.create and define messages as in the example below. There is an error from TS "No overload matches this call."

messages: [
    {
      "role": "user",
      "content": [
        {
          "type": "text",
          "text": "What is this?"
        }
}]

### To Reproduce

import OpenAI from "openai";

const openai = new OpenAI({
  apiKey: process.env.OPENAI_API_KEY,
});

const response = await openai.chat.completions.create({
  model: "gpt-4o",
  messages: [
    {
      "role": "user",
      "content": [
        {
          "type": "text",
          "text": "What is this?"
        },
        {
          "type": "image_url",
          "image_url": {
            "url": "data:image/jpeg;base64,..."
          }
        }
      ]
    },
    {
      "role": "assistant",
      "content": [
        {
          "type": "text",
          "text": "This image shows a dish composed of various cooked vegetables and what appears to be pieces of meat. The vegetables include broccoli, carrots, green beans, corn, and possibly some others. The meat is seasoned and the whole dish seems to be baked or roasted. It looks like a hearty and colorful meal!"
        }
      ]
    }
  ],
  temperature: 1,
  max_tokens: 256,
  top_p: 1,
  frequency_penalty: 0,
  presence_penalty: 0,
});

### OS

macOS

### Node version

v20.13.1

### Library version

4.47.1

**Comment by Eldo420** [comment]
### Solution Proposal for OpenAI Node.js Library Issue #867

#### Solution: Modify the Type Definition to Allow Objects for `content`

Based on the issue described, the best course of action would be to modify the type definition to allow the `content` field to accept either a string or an object. This will address the TypeScript error and offer more flexibility for users who might want to pass structured data.

---

### Implementation Steps

1. **Update Type Definitions**: 
   Modify the `ChatCompletionMessageParam` type to accept an object for `content`.

2. **Update Documentation**:
   Ensure that the changes are well-documented so users are aware of the new flexibility.

---

#### Update Type Definitions

1. Open the file where `ChatCompletionMessageParam` is defined (usually in a `types.d.ts` or similar file).
2. Modify the type definition to allow `content` to be either a string or an object.

```typescript
// Assuming types.d.ts or equivalent file
export type ChatCompletionMessageParam = {
  role: string;
  content: string | object;  // Updated to allow content as an object
};
```

#### Update the Documentation

1. Locate the documentation files (usually in a `docs` directory or within the `README.md`).
2. Add a section or update existing sections to describe the new `content` flexibility.

```markdown
### ChatCompletionMessageParam

The `ChatCompletionMessageParam` type now allows for more flexibility in the `content` field. Users can pass either a string or an object.

```typescript
import { ChatCompletionMessageParam } from 'openai';

const message1: ChatCompletionMessageParam = {
  role: 'user',
  content: "Hello, how are you?"
};

const message2: ChatCompletionMessageParam = {
  role: 'user',
  content: { text: "Hello, how are you?", meta: { timestamp: Date.now() } }
};

console.log(message1, message2);
```
```

### Testing the Changes

1. **TypeScript Compilation**:
   Ensure the code compiles without TypeScript errors.

2. **Unit Tests**:
   Add unit tests to verify that the `content` field works correctly with both strings and objects.

```typescript
// Sample unit test file, e.g., `test/ChatCompletionMessageParam.test.ts`

import { ChatCompletionMessageParam } from 'openai';

describe('ChatCompletionMessageParam', () => {
  it('should accept content as a string', () => {
    const message: ChatCompletionMessageParam = {
      role: 'user',
      content: "Hello, how are you?"
    };
    expect(typeof message.content).toBe('string');
  });

  it('should accept content as an object', () => {
    const message: ChatCompletionMessageParam = {
      role: 'user',
      content: { text: "Hello, how are you?", meta: { timestamp: Date.now() } }
    };
    expect(typeof message.content).toBe('object');
  });
});
```

---

### Submitting the Pull Request

1. **Fork the Repository**: Fork the OpenAI Node.js repository to your GitHub account.
2. **Clone the Repository**: Clone your fork to your local machine.

   ```sh
   git clone https://github.com/your-username/openai-node.git
   cd openai-node
   ```

3. **Create a New Branch**: Create a new branch for your fix.

   ```sh
   git checkout -b fix-issue-867
   ```

4. **Make the Changes**: Apply the changes listed above.

5. **Commit and Push**: Commit your changes and push them to your fork on GitHub.

   ```sh
   git add .
   git commit -m "Fix issue #867: Allow content to be an object in ChatCompletionMessageParam"
   git push origin fix-issue-867
   ```

6. **Create a Pull Request**: Go to the original repository and create a pull request with a description of the changes.

---

### Conclusion

By following these steps, you should be able to modify the OpenAI Node.js library to allow more flexible use of the `content` field in the `ChatCompletionMessageParam` type. This solution addresses the TypeScript error and enhances the library's usability. 

Feel free to let me know if you need further assistance or have any questions!

**Comment by rattrayalex** [comment]
There is a bug in your code; `role: assistant` requires `content` to be a string, not an object. 

```ts
    {
      "role": "assistant",
      "content": [
        {
          "type": "text",
          "text":
            "This image shows a dish composed of various cooked vegetables and what appears to be pieces of meat. The vegetables include broccoli, carrots, green beans, corn, and possibly some others. The meat is seasoned and the whole dish seems to be baked or roasted. It looks like a hearty and colorful meal!",
        },
      ],
    },
// should be:
    {
      "role": "assistant",
      "content": "This image shows a dish composed of various cooked vegetables and what appears to be pieces of meat. The vegetables include broccoli, carrots, green beans, corn, and possibly some others. The meat is seasoned and the whole dish seems to be baked or roasted. It looks like a hearty and colorful meal!",
    },
